draft-ietf-httpbis-message-signatures.md

```diff
@@ -1283,22 +1292,167 @@ The table below contains the initial contents of the HTTP Signature Specialty Co
+
+## Signatures do not provide confidentiality {#privacy-confidentiality}
+
+HTTP Message Signatures do not provide confidentiality of any of the information protected by the signature. The content of the HTTP message, including the value of all fields and the value of the signature itself, is presented in plaintext to any party with access to the message.
+
+To provide confidentiality at the transport level, TLS or its equivalent can be used as discussed in {{security-tls}}.
+
+## Oracles {#privacy-oracle}
+
+It is important to balance the need for providing useful feedback to developers on error conditions without providing additional information to an attacker. For example, a naive but helpful server implementation might try to indicate the required key identifier needed for requesting a resource. If someone knows who controls that key, a correlation can be made between the resource's existence and the party identified by the key. Access to such information could be used by an attacker as a means to target the legitimate owner of the resource for further attacks.
+
+## Required Content {#privacy-required}
 
-(( TODO: Should provide some recommendations on how to determine what components need to be signed for a given use case. ))
+A core design tenet of this specification is that all message components covered by the signature need to be available to the verifier in order to recreate the signature input string and verify the signature. As a consequence, if an application of this specification requires that a particular field be signed, the verifier will need access to the value of that field. 
 
-There are a number of security considerations to take into account when implementing or utilizing this specification.  A thorough security analysis of this protocol, including its strengths and weaknesses, can be found in {{WP-HTTP-Sig-Audit}}.
+For example, in some complex systems with intermediary processors this could cause the surprising behavior of an intermediary not being able to remove privacy-sensitive information from a message before forwarding it on for processing, for fear of breaking the signature. A possible mitigation for this specific situation would be for the intermediary to verify the signature itself, then modifying the message to remove the privacy-sensitive information. The intermediary can ad its own signature at this point to signal to the next destination that the incoming signature was validated, as is shown in the example in {{signature-multiple}}.
```

Comment:
typo: `The intermediary can ad its own signature` => `The intermediary can add its own signature`